### PR TITLE
fixed issue when scaling container by 1

### LINKED ISSUE #536: v3: unable to scale a single instance
When trying to scale an instance count of "1" it returns but nothing is deployed.

### PR DESCRIPTION
**Pull request by lyndaoleary**
Fixes #536 
Signed-off-by: Lynda O'Leary lyndaoleary@hotmail.com

### CODE REVIEW

File: controller/manager/manager.go [hunk 1]
```diff
@@ -244,29 +244,27 @@ func (m DefaultManager) ScaleContainer(id string, numInstances int) ScaleResult
 		return result
 	}
 
-	for i := 1; i < numInstances; i++ {
+	for i := 0; i < numInstances; i++ {
 		go func(instance int) {
-			for {
```
Comment (lyndaoleary) [inline controller/manager/manager.go:249]: this infinite loop was causing more than 1 container to be created per iteration

File: controller/manager/manager.go [hunk 2]
```diff
@@ -244,29 +244,27 @@ func (m DefaultManager) ScaleContainer(id string, numInstances int) ScaleResult
 		return result
 	}
 
-	for i := 1; i < numInstances; i++ {
```
Comment (lyndaoleary) [inline controller/manager/manager.go:247]: changing from 1 to 0 fixes the single container scaling issue.

File: controller/manager/manager.go [hunk 3]
```diff
@@ -244,29 +244,27 @@ func (m DefaultManager) ScaleContainer(id string, numInstances int) ScaleResult
...
+			if err := m.client.StartContainer(id, hostConfig); err != nil {
+				errChan <- err
+				return
+			}
+			resChan <- id
 		}(i)
 	}
 
-	for i := 1; i < numInstances; i++ {
+	for i := 0; i < numInstances; i++ {
```
Comment (lyndaoleary) [inline controller/manager/manager.go:267]: changed number of expected results to match the number of container creation requests.

File: controller/manager/manager.go (hunk 2 above)
Comment (ehazlett) [inline controller/manager/manager.go:247]: :+1:

**Comment by ehazlett** [comment]
LGTM

**Pull request merged**